### DISCUSSION TITLE
Will I get ban for using custom driver for my keyboard?

### DISCUSSION THREAD

**Discussion by okolenmi** [ppy/osu#22824] [Q&A] [answered]
I have installed a new non-oficial driver for my keyboard: [GitHub link](https://github.com/samartzidis/WinAppleKey)
Cause is it **provides very small input latency** (original is 24ms, now it's around 0-5ms). The driver can be installed only in **Windows testing mode**. I know that custom drivers can be used for cheating so I wonder if there is a chance to get ban for using my new driver.

**Comment by NotGumballer91** [comment]
No. _As far as I'm corcerned_

**Reply by okolenmi** [reply]
I don't know what methods are used to detect cheats. There is no attention to already banned players and it would be difficult to solve such a problem after the ban, so I want to make sure.

**Comment by peppy** [comment, marked as answer]
Test mode is not supported with most games. It is not supported with stable and won't be supported with lazer eventually.
